asserts/account_key.go

```diff
@@ -96,6 +104,21 @@ func (ak *AccountKey) checkConsistency(db RODatabase, acck *AccountKey) error {
 	if err != nil {
 		return err
 	}
+	// XXX: Make this unconditional once account-key assertions are required to have a name.
+	if ak.Name() != "" {
+		assertions, err := db.FindMany(AccountKeyType, map[string]string{
```

Comment:
I worry about this a little. The "check now, add some time later" style means there's a race that could result in a duplicate `name` anyway. But I'm not sure there's anything better we can do about it at this level (ideally, we want a unique constraint in a real database to enforce this).